.github/workflows/build-for-release.yml

```diff
@@ -29,52 +27,69 @@ jobs:
         echo "It must start with either 'v0.' or 'v1.'."
         echo "Please, try again."
         exit 1
-
-    # EE is always v1.x.y, OSS is always v0.x.y
-    - name: Determine the canonical version
+    - uses: actions/checkout@v3
+      with:
+        sparse-checkout: release
```

Comment:
moving the version check to a separate job from the actual build lets us check out the release code in this job, but check out the code to be built in the actual build job. This simply passes the version-parsing outputs to the other job, so the release code doesn't have to be in the commit being built.